playbooks/openshift-hosted/private/redeploy-router-certificates.yml

```diff
@@ -87,14 +100,13 @@
     - name: Add serving-cert-secret annotation to router service
       command: >
         {{ openshift_client_binary }} annotate service/router
-        service.alpha.openshift.io/serving-cert-secret-name=router-certs
+        service.alpha.openshift.io/serving-cert-secret-name=router-metrics-tls
         --config={{ router_cert_redeploy_tempdir.stdout }}/admin.kubeconfig
         -n default
     when:
     - l_router_dc.rc == 0
     - l_router_svc.rc == 0
-    - ('router-certs' in router_secrets)
-    - openshift_hosted_router_certificate is undefined
```

Comment:
Why is it being removed?

Reply:
"router-metrics-tls" secret is provided by the serving cert signer component, it does not depend on "openshift_hosted_router_certificate" which provides custom wildcard API certificate and this block is required to delete "router-metrics-tls", so the checking condition about "router-certs" secret is not matched this block.

Reply:
That makes sense, thanks!